Sources/ArcGISToolkit/Components/Offline/PreplannedMapModel.swift

```diff
@@ -130,7 +130,9 @@ class PreplannedMapModel: ObservableObject, Identifiable {
     func updateDownloadStatus(for downloadResult: Result<DownloadPreplannedOfflineMapResult, any Error>) {
         switch downloadResult {
         case .success:
-            status = .downloaded
+            withAnimation(.easeInOut) {
```

Comment:
This really shouldn't be done by the model as `withAnimation` is a SwiftUI function. I just noticed that this file imports SwiftUI, which is a huge code smell. The model should know nothing about the view; the view itself should perform the animation in response to a change in the model's state.

Reply:
Yes using `withAnimation` in the model was a workaround. The `animation<V>(_ animation: Animation?, value: V)` view modifier requires a value to change to animate, and `model.status` is not equatable. I tried using `model.status.isDownloaded` but that does not trigger an animation. Do you suggest making `model.status` equatable or another option?

Reply:
There is another option: `View.onReceive(_:perform:)`. First, revert these changes and change `import SwiftUI` to `import Combine`. Then, in the list item view, add a new download state type and state property:

```swift
private enum DownloadState {
    case notDownloaded, downloading, downloaded
    
    init(_ state: PreplannedMapModel.Status) {
        self = switch state {
        case .downloaded: .downloaded
        case .downloading: .downloading
        default: .notDownloaded
        }
    }
}

@State private var downloadState: DownloadState = .notDownloaded
```

Then, add a couple modifiers to set the initial download state and update the download state in response to the model status changes:

```swift
.onAppear {
    downloadState = .init(model.status)
}
.onReceive(model.$status) { status in
    let downloadState = DownloadState(status)
    withAnimation(
        downloadState == .downloaded ? .easeInOut : nil
    ) {
        self.downloadState = downloadState
    }
}
```

Lastly, update the code in `downloadButton` to use the new download state.